cli/command/stack/deploy_composefile.go

```diff
@@ -321,7 +321,16 @@ func deployServices(
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
 			}
-
+			if !updateOpts.QueryRegistry {
+				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
+
+				if service.Spec.TaskTemplate.Placement != nil {
+					if serviceSpec.TaskTemplate.Placement == nil {
+						serviceSpec.TaskTemplate.Placement = &swarm.Placement{}
+					}
+					serviceSpec.TaskTemplate.Placement.Platforms = service.Spec.TaskTemplate.Placement.Platforms
```

Comment:
I agree it is related, however I don't think this is the correct fix. It is not the responsibility of this code to set default values on update that it never set on create.

This looks like a bug in `ServiceUpdate()` we should not have merged that. That logic is being handled in the wrong place.